merkle/merkle.go

```diff
@@ -0,0 +1,52 @@
+package merkle
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/trace"
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/ava-labs/avalanchego/x/merkledb"
+	"github.com/ava-labs/hypersdk/utils"
+)
+
+// Generate merkle root for a set of items
+func GenerateMerkleRoot(ctx context.Context, tracer trace.Tracer, merkleItems [][]byte, consumeBytes bool) (ids.ID, merkledb.MerkleDB, error) {
+	batchOps := make([]database.BatchOp, 0, len(merkleItems))
+
+	for _, item := range merkleItems {
+		key := utils.ToID(item)
+		batchOps = append(batchOps, database.BatchOp{
+			Key:   key[:],
+			Value: item,
+		})
+	}
+
+	db, err := merkledb.New(ctx, memdb.New(), merkledb.Config{
```

Comment:
We should explicitly populate all fields IMO: https://github.com/ava-labs/hypersdk/blob/5f47812e686042d774e06613f71578b9af19c351/vm/vm.go#L207-L218

Most importantly, we should allow the user to specify a root gen concurrency: https://github.com/ava-labs/hypersdk/blob/5f47812e686042d774e06613f71578b9af19c351/vm/vm.go#L210

Reply:
`vm` seems to be meant to be at an upper level compared to `merkle` because importing `vm` in the `merkle` package creates a dependency cycle. Thus I have added a new parameter `config` with the type `merkledb.Config` so that the caller is free to provide any config they like.